hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/UpdateProcessorTest.java

```diff
@@ -108,6 +120,56 @@ public void when_keyDoesNotExist_then_doesNotCreateEntry() {
         assertThat(map).containsExactly(entry(1, 1));
     }
 
+    @Test
+    public void writeAndReadTest_success() throws IOException {
+        final var evalContextWithSubjectMock = mock(ExpressionEvalContext.class);
+        when(evalContextWithSubjectMock.getSerializationService()).thenReturn(mock());
+        final var subject = new Subject(true, singleton(mock(Principal.class)), emptySet(), emptySet());
+        when(evalContextWithSubjectMock.subject()).thenReturn(subject);
+
+        final var evalContextNoSubjectMock = mock(ExpressionEvalContext.class);
+        when(evalContextNoSubjectMock.getSerializationService()).thenReturn(mock());
+
+        final var queue = new LinkedList<>();
+        final var output = mock(ObjectDataOutput.class, (Answer<Void>) invocation -> {
+            queue.add(invocation.getArguments()[0]);
+            return null;
+        });
+        final var input = mock(ObjectDataInput.class, (Answer<Object>) invocation -> queue.poll());
+
+        final var supplier = UpdatingEntryProcessor.supplier(
+                mock(),
+                mock(),
+                mock()
+        );
+        final var writer = (UpdatingEntryProcessor) supplier.get(evalContextWithSubjectMock);
+        final var reader = (UpdatingEntryProcessor) supplier.get(evalContextNoSubjectMock);
+
+        writer.writeData(output);
+        reader.readData(input);
+
+        assertThat(reader)
+                .usingRecursiveComparison()
+                .comparingOnlyFields("rowProjectorSupplier", "valueProjectorSupplier", "arguments", "subject")
+                .ignoringFields("subject.pubCredentials", "subject.privCredentials")
+                .isEqualTo(writer);
+    }
+
+    @Test
+    public void when_unexpectedObjectTypeInInput_then_fail() {
+        final var evalContextMock = mock(ExpressionEvalContext.class);
+        when(evalContextMock.getSerializationService()).thenReturn(mock());
+
+        final var input = mock(ObjectDataInput.class, (Answer<Object>) invocation -> "some unexpected object");
+        final var supplier = UpdatingEntryProcessor.supplier(
+                mock(),
+                mock(),
+                mock()
+        );
+        final var reader = (UpdatingEntryProcessor) supplier.get(evalContextMock);
+        assertThrows(ClassCastException.class, () -> reader.readData(input));
```

Comment:
`readData` mimics java serialization - it is invoked on object constructed using no-arg constructor. Is it misuse to call it on already constructed object.

also it is not strictly necessary to test (manually) what happens when we get garbage as serialized input. That might be good task for fuzzy testing.

Reply:
Additional kudos for fuzzy testing mention

Reply:
ok. test is deleted